frontend/js/tests/user/import-listens/ImportListens.test.tsx

```diff
@@ -0,0 +1,164 @@
+import userEvent from "@testing-library/user-event";
+import { RouterProvider, createMemoryRouter } from "react-router";
+import { HttpResponse, http } from "msw";
+import { SetupServerApi, setupServer } from "msw/node";
+import {
+  renderWithProviders,
+  textContentMatcher,
+} from "../../test-utils/rtl-test-utils";
+import getSettingsRoutes from "../../../src/settings/routes";
+import { ReactQueryWrapper } from "../../test-react-query";
+import { enableFetchMocks } from "jest-fetch-mock";
+
+enableFetchMocks();
+jest.unmock("react-toastify");
+
+const user = userEvent.setup();
+const routes = getSettingsRoutes();
+
+const mockImports = [
+  {
+    import_id: 1,
+    service: "spotify",
+    created: "2025-08-12T10:00:00Z",
+    metadata: {
+      filename: "spotify.zip",
+      progress: "Your data import will start soon.",
+      status: "waiting",
+    },
+    from_date: "1970-01-01T00:00:00Z",
+    to_date: "2025-08-11T00:00:00Z",
+  },
+  {
+    import_id: 2,
+    service: "listenbrainz",
+    created: "2025-08-12T10:00:00Z",
+    metadata: {
+      filename: "listenbrainz.zip",
+      progress: "Import completed!",
+      status: "completed",
+    },
+    from_date: "2025-08-10T00:00:00Z",
+    to_date: "2025-08-12T00:00:00Z",
+  },
+];
+
+describe("ImportListensPage", () => {
+  let server: SetupServerApi;
+  let router: ReturnType<typeof createMemoryRouter>;
+
+  beforeAll(() => {
+    window.scrollTo = jest.fn();
+    window.HTMLElement.prototype.scrollIntoView = jest.fn();
+
+    server = setupServer(
+      http.post("/settings/import/", () => {
+        return HttpResponse.json({ user_has_email: true });
+      }),
+      http.get("/1/import-listens/list/", () => {
+        return HttpResponse.json(mockImports);
+      }),
+      http.post("/1/import-listens/cancel/1/", () => {
+        return HttpResponse.json({ success: true, id: 1 });
+      }),
+      http.post("/1/import-listens/1/", () => {
+        return HttpResponse.json(mockImports[0]);
+      })
+    );
+    server.listen();
+
+    router = createMemoryRouter(routes, {
+      initialEntries: ["/settings/import/"],
+    });
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+
+  it("renders submission form correctly", async () => {
+    renderWithProviders(
+      <RouterProvider router={router} />,
+      {},
+      { wrapper: ReactQueryWrapper },
+      false
+    );
+    await waitFor(() => {
+      expect(screen.getByText(/start import from/i)).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByText(/end date for import/i)).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByText(/select Service/i)).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByText(/choose a File/i)).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByRole("button", { name: /import listens/i })).toBeInTheDocument();
+      
+    });
```

Comment:
You should be able to put all these assertions inside the same waitFor block rather than each in a separate waitFor.
Same goes for the other tests.